core/vm/lfvm/interpreter.go

```diff
@@ -704,5 +737,16 @@ func step(c *context) {
 func getGasPrice(c *context) uint64 {
 	// Idea: handle static gas price in static dispatch above (saves an array lookup)
 	op := c.code[c.pc].opcode
-	return getStaticGasPrice(op)
+	return getStaticGasPrice(op, c.evm.ChainConfig().IsBerlin(c.evm.Context.BlockNumber))
+}
+
+var writeIns = []OpCode{SSTORE, LOG0, LOG1, LOG2, LOG3, LOG4, CREATE, CREATE2, SELFDESTRUCT}
+
+func isWriteInstruction(opCode OpCode) bool {
+	for _, ins := range writeIns {
+		if ins == opCode {
+			return true
+		}
+	}
```

Comment:
Using a bit mask to represent the sets should be significantly faster. Something like
```
var writeInts = getWriteInstructionMask();

func getWriteInstructionMask() *bitvector.Bitvector {
   res := &bitvector.NewBitVector(make([]byte,NUM_OPCODES/8+1), NUM_OPCODES)
   res.Set(1, SSTORE)
   ...
   res.Set(1, SELFDESTRUCT)
   return res
}

func isWriteInstruction(opCode OpCode) bool {
  return opCode < NUM_OPCODES &&  writeInts.Element(opCode) == 1
}

Reply:
done